Makefile

```diff
@@ -18,6 +18,9 @@ GATEKEEPER_VERSION = v3.10.0
 GATEKEEPER_IMAGE ?= ${RP_IMAGE_ACR}.azurecr.io/gatekeeper:$(GATEKEEPER_VERSION)
 GOTESTSUM = gotest.tools/gotestsum@v1.11.0
 
+# Golang version go mod tidy compatibility
+GOLANG_VERSION ?= 1.20
```

Comment:
Should we specify 1.20.12? (Still finding my feet with Go, I have no idea how strict it is with versions)

For reference: https://github.com/Azure/ARO-RP/pull/3548

Reply:
go mod tidy is expecting an x.y version, thus we can't do that.

> go mod tidy -compat=1.20.12
invalid value "1.20.12" for flag -compat: expecting a Go version like "1.20"
usage: go mod tidy [-e] [-v] [-x] [-go=version] [-compat=version]
Run 'go help mod tidy' for details.
make: *** [Makefile:283: go-tidy] Error 2

Reply:
Let's just double check this is the version we need. I remember some CVE issues so even though we just moved to 1.20 we need to make sure we have the version with the xnet fixes in.

Reply:
Go 1.20 is the best version we can support right now, we'll definitely need to move to 1.22 soon though. My understanding is that we have many underlying components we have to upgrade as well if we do, so 1.20 min is our first best step towards compliance.